src/dockerRunner.ts

```diff
@@ -72,39 +72,58 @@ export class DockerRunner extends TerminalBaseRunner {
             const msgOption: vscode.MessageOptions = { modal: false };
             const msgItem: vscode.MessageItem = { title: 'Ok' };
 
-            const cancelItem: vscode.MessageItem = { title: "Open File" };
-            const promptMsg = 'Using credential file at ' + utilities.getCredentialsFile();
-            vscode.window.showInformationMessage(promptMsg, msgOption, msgItem, cancelItem).then(response => {
-                if (response === cancelItem) {
-                    vscode.workspace.openTextDocument(utilities.getCredentialsFile()).then(doc => {
-                        vscode.window.showTextDocument(doc);
-                    });
-                }
-            });
-            TerminalExecutor.runInTerminal(initCmd, Constants.AnsibleTerminalName + ' ' + Option.docker, true, subCmds, 180, false, function (terminal, interval) {
-                if (terminal) {
-                    require('child_process').exec('docker ps --filter name=' + terminalId, (err, stdout, stderr) => {
-                        if (err || stderr) {
-                            console.log('err: ' + err + ' ' + stderr);
-                            return;
-                        }
-                        if (stdout) {
-                            // check if docker container is up
-                            if (stdout && stdout.indexOf('Up ') > -1) {
-
-                                // then send other commands to terminal
-                                for (let text of subCmds) {
-                                    terminal.sendText(text);
-                                }
-                                terminal.show();
-                                if (interval) {
-                                    clearInterval(interval);
-                                }
+            if (!utilities.isCredentialConfigured()) {
+                const cancelItem: vscode.MessageItem = { title: "Not Now" };
+                const promptMsg = 'Please configure cloud credentials at ' + utilities.getCredentialsFile() + ' for first time';
+                
+                vscode.window.showWarningMessage(promptMsg, msgOption, msgItem, cancelItem).then(response => {
+                    vscode.workspace.getConfiguration('ansible').update('credentialsConfigured', true);
+                    if (response === msgItem) {
+                        vscode.workspace.openTextDocument(utilities.getCredentialsFile()).then(doc => {
+                            vscode.window.showTextDocument(doc);
```

Comment:
so credentialConfigured is set to true after user opens the credentials document?
How about doing it when user actually saves the document?
we could intercept save event, and then do some verification perhaps?

Reply:
good point, let me do the change.

Reply:
second thought, i remembered why i not update it when user change file, let me sync with you offline